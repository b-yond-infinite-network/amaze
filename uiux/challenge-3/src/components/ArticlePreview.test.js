import ArticlePreview from './ArticlePreview';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from './../store'

describe('ArticlePreview Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    beforeAll(() => {

        mockStore = configureMockStore(middlewares)({});

        const props = {
            article: {
                title: 'Untitled Title',
                slug: 'a-title-7ua6fd',
                body: '# Nic\nbylo nebylo',
                createdAt: '2019-06-29T18:36:23.216Z',
                updatedAt: '2019-06-29T18:37:11.601Z',
                tagList: [
                    'nothing',
                    'nic'
                ],
                description: 'o niƒçem',
                author: {
                    username: 'tester96',
                    bio: 'b',
                    image: '',
                    following: false
                },
                favorited: false,
                favoritesCount: 1
            }
        };

        tree = <Provider store={mockStore}>
            <MemoryRouter initialEntries={["/"]}>
                <ArticlePreview {...props} />
            </MemoryRouter>
        </Provider>;
        wrapper = mount(tree);
        console.log(wrapper.debug());
        render = renderer.create(tree).toJSON();

    });

    it('Should match the snapshot', () => {
        expect(render).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const component = wrapper.find('#article-preview');
        expect(component.length).toBe(1);
    });

});