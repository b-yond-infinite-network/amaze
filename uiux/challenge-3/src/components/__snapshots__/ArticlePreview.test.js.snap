// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreview Component Should match the snapshot 1`] = `
<div
  className="article-preview"
  id="article-preview"
>
  <div
    className="article-meta"
  >
    <a
      href="/@tester96"
      onClick={[Function]}
    >
      <img
        alt="tester96"
        src=""
      />
    </a>
    <div
      className="info"
    >
      <a
        className="author"
        href="/@tester96"
        onClick={[Function]}
      >
        tester96
      </a>
      <span
        className="date"
      >
        Sat Jun 29 2019
      </span>
    </div>
    <div
      className="pull-xs-right"
    >
      <button
        className="btn btn-sm btn-outline-primary"
        onClick={[Function]}
      >
        <i
          className="ion-heart"
        />
         
        1
      </button>
    </div>
  </div>
  <a
    className="preview-link"
    href="/article/a-title-7ua6fd"
    onClick={[Function]}
  >
    <h1>
      Untitled Title
    </h1>
    <p>
      o niƒçem
    </p>
    <span>
      Read more...
    </span>
    <ul
      className="tag-list"
    >
      <li
        className="tag-default tag-pill tag-outline"
      >
        nothing
      </li>
      <li
        className="tag-default tag-pill tag-outline"
      >
        nic
      </li>
    </ul>
  </a>
</div>
`;
